core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -54,25 +58,45 @@ private HeartbeatScheduler() {} // to prevent initialization
   public static void addTimer(ScheduledTimer timer) {
     Preconditions.checkNotNull(timer);
     try (LockResource r = new LockResource(sLock)) {
+      Preconditions.checkState(!sTimers.containsKey(timer.getThreadName()),
+          "The timer for thread %s is already waiting to be scheduled", timer.getThreadName());
```

Comment:
This file is used for testing only and it should be impossible to trigger these preconditions, so I think it's better to leave the text here so that it can also serve as documentation.